ontology-engine/graph-engine_2.11/src/main/scala/org/sunbird/graph/nodes/DataNode.scala

```diff
@@ -275,16 +275,22 @@ object DataNode {
   def search(request: Request)(implicit oec: OntologyEngineContext, ec: ExecutionContext): Future[List[Node]] = {
     list(request, Some(request.getObjectType)).map(nodeList => {
       validateNodeList(request, nodeList)
+      val fields: List[String] = Optional.ofNullable(request.get("fields").asInstanceOf[util.List[String]]).orElse(new util.ArrayList[String]()).toList
       val extPropNameList = DefinitionNode.getExternalProps(request.getContext.get("graph_id").asInstanceOf[String], request.getContext.get("version").asInstanceOf[String], request.getContext().get("schemaName").asInstanceOf[String])
-      populateExternalProperties(nodeList.asScala.toList, request, extPropNameList)
+      if (CollectionUtils.isEmpty(fields) && CollectionUtils.isNotEmpty(extPropNameList))
+        populateExternalProperties(nodeList.asScala.toList, extPropNameList, request, extPropNameList)
+      else if (CollectionUtils.isNotEmpty(extPropNameList) && fields.exists(field => extPropNameList.contains(field)))
```

Comment:
If `fields` is not empty and not matching with `extPropNameList` list then, we should not call `populateExternalProperties`. Please try to optimise the DB calls here.